astropy/coordinates/builtin_frames/utils.py

```diff
@@ -376,3 +376,40 @@ def prepare_earth_position_vel(time):
         )
 
     return earth_pv, earth_heliocentric
+
+
+def offset_sun_from_barycenter(time, include_velocity=False, reverse=False):
```

Comment:
It is very annoying that representations cannot be simply negated, as that would avoid `reverse` here. In principle, that is not difficult to solve, but would be a new feature, so not so helpful here (but see #10987). Looking at `solar_system`, I also think that, really, `get_body_barycentric` itself should have the `include_velocity` option. We could consider calling it `get_sun_barycentric(time, include_velocity=False, reverse=False)` here (and then later deprecate reverse).